src/test/java/updated/game/abstractClassImpl/TicTacToe.java

```diff
@@ -0,0 +1,158 @@
+        System.out.println("******************************");
+        System.out.printf("%20s", "Welcome \uD83D\uDC4B \uD83D\uDC4B \n");
+        //
+        System.out.println("⚠️Press Q to Quit on going old.game.⚠️ \n\n");
+        System.out.println("The board will be as");
+        StringBuilder cliResult = new StringBuilder();
+        for (int i = 0; i < Board.BOARD_SIZE; i++) {
+            for (int j = 0; j < Board.BOARD_SIZE; j++) {
+                if (j != (Board.BOARD_SIZE - 1))
+                    cliResult.append("X").append("|");
+                else
+                    cliResult.append('X').append("\n");
+            }
+            if (i != (Board.BOARD_SIZE - 1))
+                cliResult.append("-".repeat(TicTacToeBoard.BOARD_SIZE*TicTacToeBoard.BOARD_SIZE)).append("\n");
+        }
+
+        cliResult.append("\nYou are suppose to provide input as number of box you want to place your symbol\n");
+        int k = 1;
+        for (int i = 0; i < Board.BOARD_SIZE; i++) {
+            for (int j = 0; j < Board.BOARD_SIZE; j++) {
+                if (j != (Board.BOARD_SIZE - 1))
+                    cliResult.append(k).append(" | ");
+                else
+                    cliResult.append(k).append("\n");
+                k++;
+            }
+            if (i != (Board.BOARD_SIZE - 1))
+                cliResult.append("-".repeat(TicTacToeBoard.BOARD_SIZE*TicTacToeBoard.BOARD_SIZE)).append("\n");
+        }
+        cliResult.append("******************************\n");
+        cliResult.append("\n******************************\n");
+        System.out.println(cliResult.toString());
+
+        human.getUserNameFromCli();
+        human.getSymbolFromUser();
+        computer.setSymbol((human.getSymbol() == 'O') ? 'X' : 'O');
+        human.printDetails();
+        System.out.println("******************************");
+    }
+
+    @Override
+    public void play() {
+        this.printInstructions();
+        do{
+            System.out.println("\n\nLet's Begin \uD83C\uDFB2");
+            System.out.println("\n\nCurrent Status");
+            this.board.printBoard();
+
+            while (!isGameOver(human.getNumberOfMoves(),computer.getNumberOfMoves())) {
+                System.out.printf("========== %s's Turn ==========\n", human.getGamerName());
+                List<Integer> rowAndCol = human.getInputFromUser(this);
+                board.userMove(human, rowAndCol.get(0), rowAndCol.get(1));
+                System.out.println("Board");
+                this.board.printBoard();
+                if (isGameOver(human.getNumberOfMoves(),computer.getNumberOfMoves()) || ((human.getNumberOfMoves() + computer.getNumberOfMoves()) == (TicTacToeBoard.BOARD_SIZE * TicTacToeBoard.BOARD_SIZE)) ) {
+                    System.out.println("IN IF");
+                    break;
+                }
+                System.out.println("========== Computer's Turn ==========");
+                rowAndCol = computer.getInputFromUser(this);
```

Comment:
How about incorporating common logic from user and computer move into different method